sympy/integrals/manualintegrate.py

```diff
@@ -211,9 +211,6 @@ def test_subterm(u, u_diff):
         if u_diff == 0:
             return False
         substituted = integrand / u_diff
-        if symbol not in substituted.free_symbols:
-            # replaced everything already
-            return False
```

Comment:
So the idea is this previously wouldn't allow u-substitutions where the du was just the entire integrand, as in the example? Such as substitution seems unhelpful from a steps point of view, but it's also useful if it does mean it can compute more integrals.

Reply:
Exactly.